### PR TITLE
Added product listing page

### PR DESCRIPTION
**Pull request by jeetsdev**
Added product listing page with filter 

- All products 
- Filter section
  -  Filter by price 
  - Filter by Category 
  - Filter by ratings
  - Sort by price

[Preview](https://deploy-preview-2--thriving-gumption-494d22.netlify.app/)

### CODE REVIEW

File: src/components/Filter/Filter.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        <div className="filter__sec-sort">
+            <h3>Sort by</h3>
+            <form action="" className="center__flex flex__dir-col">
+                <div>
+                    <input type="radio" name="sort-by-price" id="low-to-high" onChange={() => filterDispatch({ type: `SORT_BY_PRICE`, payload: "LOW_TO_HIGH" })} />
+                    <label htmlFor="low-to-high"> Price - low to high  </label>
+                </div>
+                <div>
+                    <input type="radio" name="sort-by-price" id="high-to-low" onChange={() => filterDispatch({ type: `SORT_BY_PRICE`, payload: "HIGH_TO_LOW" })} />
+                    <label htmlFor="high-to-low"> Price - high to low  </label>
```
Comment (JowelTisso) [inline src/components/Filter/Filter.jsx]: You can export the 'SORT_BY_PRICE' and similar types to constant variable

File: src/contexts/product-context.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    useEffect(() => {
+        (async () => {
+            try {
+                let res = await axios.get("/api/products")
+                setProductData(res.data.products)
+            } catch (error) {
+                console.log(error.message)
+            }
+        })()
+        // console.log(productData);
```
Comment (JowelTisso) [inline src/contexts/product-context.jsx]: Remember to remove console logs in production

**Review by JowelTisso — APPROVED** [review]
Suggestions: You can use links for products
Rest is good.

**Review by manismk — APPROVED** [review]
Good naming convention. Easy to understand code.
Suggestion: You use constants for action types.

**Comment by manismk** [comment]
Tried filter and found an issue. After clicking clear, the products are getting reset but still, the inputs are in the selected state.


image

**Pull request merged**